mesonbuild/build.py

```diff
@@ -1676,6 +1676,9 @@ def get_generated_sources(self):
     def type_suffix(self):
         return "@cus"
 
+    def get_index_protocol(self, index):
+        return CustomTargetIndex(self, self.outputs[index])
```

Comment:
Another way to do this would be to implement `__getitem__` and implement `__setitem__` and `__delitem__` to raise `AssertionError`s. That would mean that these objects would also be iterable from Python, but perhaps that's a good thing.

Reply:
I wasn't sure. I thought about just implementing `__getitem__` but I feel like that's a bit of a layering violation (I think that meson's lack of a ListHolder is also a bit of a layering violation)